crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -105,6 +106,99 @@ pub(super) const END_EXPR_SET: TokenSet = TokenSet::new([
 /// Tokens that can appear at the end of a sequence.
 const END_SEQUENCE_SET: TokenSet = END_EXPR_SET.remove(TokenKind::Comma);
 
+/// Represents the expression parsing context.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(super) struct ExpressionContext(ExpressionContextFlags);
```

Comment:
Nit: I would move this to the end of the file, considering that it is an implementation detail (what I'm most interested when jumping to this file is to see how expression parsing works)